DenzelleJinx/ECSE-428-Fall-2024
# FE/BE - Create Listing

## Sprint B Week 2 - Backlog item 8 - FE/BE - Create Listing

Implements the create listing feature on the frontend and ensures proper connection with the backend.

Works with the login system.

TO TEST:
1. Create a new landlord user using the signup component in frontend
2. Log in. You should see a message "user logged in successfully" in the console
3. Go to create listing page. The fields are already filled with valid sample data to make your life easier. Feel free to change the values
4. Click on "Create listing". If the values are valid, a success message will appear right above the button. The returned listing DTO is printed to the console
5. Check that the listing exists in the database

Notes:
* I left TODO comments in `CreateListing.js`
* If create listing doesn't work, try dropping and re-creating the database, then restarting the backend:
    ```sql
    DROP DATABASE house_it;
    CREATE DATABASE house_it;
    ```
